src/hashmap/cursor.rs

```diff
@@ -112,15 +112,22 @@ pub(crate) trait CursorReadOps<K: Clone + Hash + Eq + Debug, V: Clone> {
         K: Borrow<Q>,
         Q: Hash + Eq,
     {
-        // Search for and return if a value exists at key.
-        let rref = self.get_root_ref();
-        rref.get_ref(h, k)
-            // You know, I don't even want to talk about the poor life decisions
-            // that lead to this code existing.
-            .map(|v| unsafe {
-                let x = v as *const V;
-                &*x as &V
-            })
+        let mut node = self.get_root_ref();
+        loop {
```

Comment:
Make this a while or for loop with a max depth counter. If we hit max depth, then panic!();. Max depth could be 1024 or similar, as at this size the tree would be ... uhh ... huge.